src/Aquifer.AI/OpenAiTranslationService.cs

```diff
@@ -0,0 +1,194 @@
+﻿using System.Text;
+using System.Text.RegularExpressions;
+using Aquifer.Common.Clients;
+using Aquifer.Common.Utilities;
+using OpenAI.Chat;
+
+namespace Aquifer.AI;
+
+public interface ITranslationService
+{
+    /// <summary>
+    /// Translates plain text.
+    /// Note: It's up to the caller to guarantee that the text length is not too long.
+    /// </summary>
+    public Task<string> TranslateTextAsync(
+        string text,
+        (string Iso6393Code, string EnglishName) destinationLanguage,
+        CancellationToken cancellationToken);
+
+    /// <summary>
+    /// Translates HTML content while preserving the HTML tags.
+    /// Note: Translations are done on each individual paragraph in order to avoid hitting length limitations.
+    /// </summary>
+    public Task<string> TranslateHtmlAsync(
+        string html,
+        (string Iso6393Code, string EnglishName) destinationLanguage,
+        CancellationToken cancellationToken);
+}
+
+public class OpenAiOptions
+{
+    public required string Model { get; init; }
+}
+
+public sealed class OpenAiTranslationOptions
+{
+    public string? TextTranslationPromptOverride { get; init; }
+    public string? HtmlTranslationBasePromptOverride { get; init; }
+    public IReadOnlyDictionary<string, string>? LanguageSpecificPromptAppendixByLanguageIso6393CodeMap { get; init; }
+}
+
+public sealed class OpenAiChatCompletionException(string message, string prompt, string text)
+    : Exception(message)
+{
+    public string Prompt { get; } = prompt;
+    public string Text { get; } = text;
+}
+
+public sealed partial class OpenAiTranslationService : ITranslationService
+{
+    private const int _maxContentLength = 5_000;
+    private const int _maxParallelizationForSingleTranslation = 5;
+
+    private readonly ChatClient _chatClient;
+    private readonly OpenAiTranslationOptions _options;
+
+    public OpenAiTranslationService(
+        OpenAiTranslationOptions openAiTranslationOptions,
+        OpenAiOptions openAiOptions,
+        IAzureKeyVaultClient keyVaultClient)
+    {
+        _options = openAiTranslationOptions;
+
+        // TODO Inject a ChatClient instead of building one here. This will require changing how we fetch key vault secrets.
+        const string openAiApiKeySecretName = "OpenAiApiKey";
```

Comment:
Currently the KeyVault secret names are all in a constants file only accessible by Aquifer.API.  I want to look into doing KeyVault secrets differently, anyway, so I didn't bother to refactor those constants to a shared location.